server/src/main/java/org/apache/iotdb/db/mpp/plan/statement/crud/InsertRowsOfOneDeviceStatement.java

```diff
@@ -136,4 +136,25 @@ public long getMinTime() {
   public Object getFirstValueOfIndex(int index) {
     throw new NotImplementedException();
   }
+
+  @Override
+  public InsertBaseStatement split() {
+    boolean needSplit = false;
+    for (InsertRowStatement child : this.insertRowStatementList) {
+      if (child.isNeedSplit()) {
+        needSplit = true;
+      }
+    }
```

Comment:
We can record the index and start split from the recorded index.

Reply:
But children who don't need to split still need to be collected, so I think that the time complexity is consistent and it uses less space.